mpqp/noise/noise_model.py

```diff
@@ -337,8 +337,134 @@ class PhaseFlip(NoiseModel):
     def to_kraus_representation(self) -> KrausRepresentation: ...
 
 
+@typechecked
 class AmplitudeDamping(NoiseModel):
-    """3M-TODO"""
+    """Class representing the amplitude damping noise channel, which can model both
+    the standard and generalized amplitude damping processes. It can be applied
+    to a single qubit and depends on two parameters: the decay rate `gamma` and the
+    probability of excitation `prob`.
+
+    Args:
+        gamma: Decaying rate of the amplitude damping noise channel.
+        prob: Probability of excitation in the generalized amplitude damping noise channel
+            When set to 1, indicating standard amplitude damping.
+        targets: List of qubit indices affected by this noise.
+        gates: List of :class:`Gates<mpqp.core.instruction.gates.gate.Gate>`
+            affected by this noise.
+
+    Raises:
+        ValueError: When the gamma or prob parameters are outside of the expected interval [0, 1].
+        ValueError: When no target qubits are specified.
+
+    Examples:
+        >>> circuit = QCircuit([H(i) for i in range(3)])
+        >>> ad1 = AmplitudeDamping(0.2, 0, [0])
+        >>> ad2 = AmplitudeDamping(0.4, 0.1, [1, 2])
+        >>> ad3 = AmplitudeDamping(0.1, 1, [0, 1, 2])
+        >>> ad4 = AmplitudeDamping(0.7, targets=[0, 1])
+        >>> circuit.add([ad1, ad2, ad3, ad4])
+        >>> print(circuit)
+             ┌───┐
+        q_0: ┤ H ├
+             ├───┤
+        q_1: ┤ H ├
+             ├───┤
+        q_2: ┤ H ├
+             └───┘
+        NoiseModel:
+            AmplitudeDamping(gamma=0.2, prob=0, targets=[0])
+            AmplitudeDamping(gamma=0.4, prob=0.1, targets=[1, 2])
+            AmplitudeDamping(gamma=0.1, prob=1, targets=[0, 1, 2])
+            AmplitudeDamping(gamma=0.7, prob=1, targets=[0, 1])
+
+    """
+
+    def __init__(
+        self,
+        gamma: float,
+        prob: float = 1,
```

Comment:
You should use other type than ``float`` if you give int values (like 0 or 1) for prob. I would advise using Real (from numbers)

Reply:
float works fine, and if you give 0 the error was coming from Braket when to_other_language, but with casting, it works fine